docs/write-audit-log-filter-definitions.md

```diff
@@ -0,0 +1,301 @@
+
+Prerequisites:
+* Percona Server for MySQL with audit log filter plugin installed
+
+* `AUDIT_ADMIN` privilege
+
+* Basic understanding of JSON syntax
+
+## Quick start (5 minutes)
+
+### Step 1: Create a filter
+```sql
+SET @my_filter = '{
+  "filter": {
+    "class": [
+      { "name": "connection" }
+    ]
+  }
+}';
+SELECT audit_log_filter_set_filter('my_filter', @my_filter);
+```
+
+### Step 2: Assign to user
+```sql
+SELECT audit_log_filter_set_user('user1'@'%', 'my_filter');
+```
+
+### Step 3: Test
+Connect as 'user1', run a query, and check `/var/lib/mysql/audit.log`.
+
+## JSON filter structure
+
+Every filter follows this pattern:
+
+```json
+{
+  "filter": {
+    "class": [
+      {
+        "name": "connection",
+        "log": true
+      }
+    ]
+  }
+}
+```
+
+Event types you can filter:
+
+* `"connection"` - User logins/logouts (track who accesses the database)
+
+* `"general"` - SQL queries and commands (monitor all database activity)
+
+* `"table_access"` - Table operations (SELECT, INSERT, UPDATE, DELETE)
+
+* `"query"` - Query execution details (debug slow queries)
+
+Optional fields:
+
+* `"user"` - Array of usernames to track
+
+* `"host"` - Array of client IPs to track  
+
+* `"event"` - Array of specific event subclasses to track
+
+* `"log"` - Set to `true` or `false` to enable/disable logging for this class
+
+## Common patterns
+
+Log everything (good for testing):
+```json
+{
+  "filter": {
+    "class": [
+      { "name": "connection" },
+      { "name": "general" },
+      { "name": "table_access" }
+    ]
+  }
+}
+```
+
+Track specific users only:
+```json
+{
+  "filter": {
+    "class": [
+      {
+        "name": "table_access",
+        "event": [
+          { "name": "insert" },
+          { "name": "update" },
+          { "name": "delete" }
+        ]
+      }
+    ]
+  }
+}
+```
+
+Monitor data changes only:
+```json
+{
+  "filter": {
+    "class": [
+      {
+        "name": "table_access",
+        "event": [
+          { "name": "insert" },
+          { "name": "update" },
+          { "name": "delete" }
+        ]
+      }
+    ]
+  }
+}
+```
+
+Track specific operations:
+```json
+{
+  "filter": {
+    "class": [
+      {
+        "name": "table_access",
+        "event": ["insert", "update", "delete"]
+      }
+    ]
+  }
+}
+```
+
+## Create and assign a filter
+
+Step 1: Create the filter
+```sql
+SET @my_filter = '{
+  "filter": {
+    "class": [
+      {
+        "name": "connection",
+        "log": true
+      }
+    ]
+  }
+}';
+SELECT audit_log_filter_set_filter('user_connections', @my_filter);
+```
+
+Step 2: Assign to users
+```sql
+SELECT audit_log_filter_set_user('user1'@'%', 'user_connections');
+SELECT audit_log_filter_set_user('user2'@'localhost', 'user_connections');
+```
+
+Step 3: Test your filter
+
+1. Connect as 'user1' and run: `SELECT 1;`
+
+2. Check the audit log: `tail -f /var/lib/mysql/audit.log`
+
+3. You should see JSON entries for the connection and query
```

Comment:
For the above filter, connect and disconnect will be logged in the audit log. Select query will not be logged. The audit log file will contain JSON entries only if the audit log format is JSON, the default format is New style XML.